File: com/problem/java/hackerrank/ConstantTimeFizzBuzz.java
package com.problem.java.hackerrank;

public class ConstantTimeFizzBuzz {
  public static void main(String[] args) {
    String s = "";
    int case3 = 0, case5 = 0;

    for (int i = 1; i <= 100; i++) {
      case3++;
      case5++;
      if (case3 == 3) {
        s += "fizz";
        case3 = 0;
      } else if (case5 == 5) {
        s+= "buzz";
        case5 = 0;
      }
      if (s.length() == 0)
        System.out.println(i);
      else
        System.out.println(s);
      s = "";
    }

  }
}

File: bin/com/problem/go/hackerrank/constantTimeFizzBuzz.go
package main

import "fmt"

// constant time O(1)
func main() {
  s := ""
  var case3, case5 int

  for i:=1; i<= 100; i++ {
    case3++
    case5++

    if case3==3 && case5==5 {
      s+="fizzBuzz"
      case3=0
      case5=0
    }
    if case3==3 {
      s+="fizz"
      case3=0
    }  
    if case5==5 {
      s+="buzz"
      case5=0
    } 

    if len(s)==0 {
      fmt.Println(i)
    } else {
      fmt.Println(s)
    }
    s=""
  }
}
